diff -ur libgda-3.1.5.orig/libgda/dir-blob-op.c libgda-3.1.5/libgda/dir-blob-op.c
--- libgda-3.1.5.orig/libgda/dir-blob-op.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/dir-blob-op.c	2012-06-29 12:53:05.243999205 +0200
@@ -28,7 +28,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 
 struct _GdaDirBlobOpPrivate {
 	gchar *complete_filename;
diff -ur libgda-3.1.5.orig/libgda/gda-column.c libgda-3.1.5/libgda/gda-column.c
--- libgda-3.1.5.orig/libgda/gda-column.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-column.c	2012-06-29 12:53:05.237332538 +0200
@@ -23,8 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-column.h>
 #include <string.h>
 #include "gda-marshal.h"
diff -ur libgda-3.1.5.orig/libgda/gda-column.h libgda-3.1.5/libgda/gda-column.h
--- libgda-3.1.5.orig/libgda/gda-column.h	2008-07-20 11:43:15.000000000 +0200
+++ libgda-3.1.5/libgda/gda-column.h	2012-06-29 12:53:05.237332538 +0200
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 #include <libgda/gda-value.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
diff -ur libgda-3.1.5.orig/libgda/gda-command.c libgda-3.1.5/libgda/gda-command.c
--- libgda-3.1.5.orig/libgda/gda-command.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-command.c	2012-06-29 12:53:05.243999205 +0200
@@ -21,9 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-command.h>
 
 GType
diff -ur libgda-3.1.5.orig/libgda/gda-command.h libgda-3.1.5/libgda/gda-command.h
--- libgda-3.1.5.orig/libgda/gda-command.h	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-command.h	2012-06-29 12:53:05.243999205 +0200
@@ -25,8 +25,7 @@
 #define __GDA_COMMAND_H__
 
 #include <glib-object.h>
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
diff -ur libgda-3.1.5.orig/libgda/gda-config.c libgda-3.1.5/libgda/gda-config.c
--- libgda-3.1.5.orig/libgda/gda-config.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-config.c	2012-06-29 12:53:05.240665871 +0200
@@ -38,7 +38,7 @@
 #include <sys/stat.h>
 #ifdef HAVE_FAM
 #include <fam.h>
-#include <glib/giochannel.h>
+#include <glib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
diff -ur libgda-3.1.5.orig/libgda/gda-config.h libgda-3.1.5/libgda/gda-config.h
--- libgda-3.1.5.orig/libgda/gda-config.h	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-config.h	2012-06-29 12:53:05.240665871 +0200
@@ -23,7 +23,7 @@
 #ifndef __GDA_CONFIG_H__
 #define __GDA_CONFIG_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-parameter.h>
 
diff -ur libgda-3.1.5.orig/libgda/gda-data-model-array.c libgda-3.1.5/libgda/gda-data-model-array.c
--- libgda-3.1.5.orig/libgda/gda-data-model-array.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-data-model-array.c	2012-06-29 12:53:05.240665871 +0200
@@ -22,7 +22,7 @@
  */
 
 #include <string.h>
-#include <glib/garray.h>
+#include <glib.h>
 #include <libgda/gda-data-model-array.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-row.h>
diff -ur libgda-3.1.5.orig/libgda/gda-data-model.c libgda-3.1.5/libgda/gda-data-model.c
--- libgda-3.1.5.orig/libgda/gda-data-model.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-data-model.c	2012-06-29 12:53:05.253999204 +0200
@@ -23,7 +23,7 @@
  */
 
 #include <glib/gi18n-lib.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-data-model-private.h>
 #include <libgda/gda-data-model-extra.h>
diff -ur libgda-3.1.5.orig/libgda/gda-data-model-dir.c libgda-3.1.5/libgda/gda-data-model-dir.c
--- libgda-3.1.5.orig/libgda/gda-data-model-dir.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-data-model-dir.c	2012-06-29 12:53:05.250665871 +0200
@@ -21,7 +21,7 @@
 
 #include <string.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-data-model-dir.h>
 #include <libgda/gda-data-model-extra.h>
diff -ur libgda-3.1.5.orig/libgda/gda-data-model-filter-sql.c libgda-3.1.5/libgda/gda-data-model-filter-sql.c
--- libgda-3.1.5.orig/libgda/gda-data-model-filter-sql.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-data-model-filter-sql.c	2012-06-29 12:53:05.257332538 +0200
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gdataset.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/libgda/gda-data-model-hash.c libgda-3.1.5/libgda/gda-data-model-hash.c
--- libgda-3.1.5.orig/libgda/gda-data-model-hash.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-data-model-hash.c	2012-06-29 12:53:05.250665871 +0200
@@ -23,7 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/ghash.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-data-model-hash.h>
 #include <libgda/gda-data-model-extra.h>
diff -ur libgda-3.1.5.orig/libgda/gda-dict.c libgda-3.1.5/libgda/gda-dict.c
--- libgda-3.1.5.orig/libgda/gda-dict.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-dict.c	2012-06-29 12:53:05.233999205 +0200
@@ -19,7 +19,7 @@
 
 #include <string.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include "gda-dict.h"
 #include "gda-dict-private.h"
 #include "gda-marshal.h"
diff -ur libgda-3.1.5.orig/libgda/gda-init.c libgda-3.1.5/libgda/gda-init.c
--- libgda-3.1.5.orig/libgda/gda-init.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-init.c	2012-06-29 12:53:05.233999205 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <libgda/libgda.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/libgda/gda-log.c libgda-3.1.5/libgda/gda-log.c
--- libgda-3.1.5.orig/libgda/gda-log.c	2008-07-20 11:43:15.000000000 +0200
+++ libgda-3.1.5/libgda/gda-log.c	2012-06-29 12:53:05.250665871 +0200
@@ -24,10 +24,7 @@
 #include <syslog.h>
 #endif
 #include <time.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gutils.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
 
diff -ur libgda-3.1.5.orig/libgda/gda-log.h libgda-3.1.5/libgda/gda-log.h
--- libgda-3.1.5.orig/libgda/gda-log.h	2008-07-20 11:43:15.000000000 +0200
+++ libgda-3.1.5/libgda/gda-log.h	2012-06-29 12:53:05.250665871 +0200
@@ -20,8 +20,7 @@
 #if !defined(__gda_log_h__)
 #  define __gda_log_h__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <time.h>
 
 G_BEGIN_DECLS
diff -ur libgda-3.1.5.orig/libgda/gda-quark-list.c libgda-3.1.5/libgda/gda-quark-list.c
--- libgda-3.1.5.orig/libgda/gda-quark-list.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-quark-list.c	2012-06-29 12:53:05.240665871 +0200
@@ -22,10 +22,7 @@
  */
 
 #include <libgda/gda-quark-list.h>
-#include <glib/ghash.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 struct _GdaQuarkList {
 	GHashTable *hash_table;
diff -ur libgda-3.1.5.orig/libgda/gda-quark-list.h libgda-3.1.5/libgda/gda-quark-list.h
--- libgda-3.1.5.orig/libgda/gda-quark-list.h	2008-07-20 11:43:15.000000000 +0200
+++ libgda-3.1.5/libgda/gda-quark-list.h	2012-06-29 12:53:05.240665871 +0200
@@ -24,8 +24,7 @@
 #ifndef __GDA_QUARK_LIST_H__
 #define __GDA_QUARK_LIST_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 
 G_BEGIN_DECLS
diff -ur libgda-3.1.5.orig/libgda/gda-row.h libgda-3.1.5/libgda/gda-row.h
--- libgda-3.1.5.orig/libgda/gda-row.h	2008-07-20 11:43:15.000000000 +0200
+++ libgda-3.1.5/libgda/gda-row.h	2012-06-29 12:53:05.250665871 +0200
@@ -27,7 +27,7 @@
 
 #include <glib-object.h>
 #include <libgda/gda-column.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
diff -ur libgda-3.1.5.orig/libgda/gda-server-operation.c libgda-3.1.5/libgda/gda-server-operation.c
--- libgda-3.1.5.orig/libgda/gda-server-operation.c	2008-08-04 13:16:26.000000000 +0200
+++ libgda-3.1.5/libgda/gda-server-operation.c	2012-06-29 12:53:05.270665870 +0200
@@ -21,7 +21,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-marshal.h>
 #include <libgda/gda-dict.h>
 #include <libgda/gda-dict-type.h>
diff -ur libgda-3.1.5.orig/libgda/gda-server-provider.c libgda-3.1.5/libgda/gda-server-provider.c
--- libgda-3.1.5.orig/libgda/gda-server-provider.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-server-provider.c	2012-06-29 12:53:05.253999204 +0200
@@ -22,7 +22,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-server-provider-private.h>
diff -ur libgda-3.1.5.orig/libgda/gda-util.c libgda-3.1.5/libgda/gda-util.c
--- libgda-3.1.5.orig/libgda/gda-util.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-util.c	2012-06-29 12:53:05.247332538 +0200
@@ -27,10 +27,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <glib/gfileutils.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
diff -ur libgda-3.1.5.orig/libgda/gda-util.h libgda-3.1.5/libgda/gda-util.h
--- libgda-3.1.5.orig/libgda/gda-util.h	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-util.h	2012-06-29 12:53:05.247332538 +0200
@@ -24,8 +24,7 @@
 #ifndef __GDA_UTIL_H__
 #define __GDA_UTIL_H__
 
-#include <glib/ghash.h>
-#include <glib/glist.h>
+#include <glib.h>
 #include "gda-parameter.h"
 #include "gda-row.h"
 #include "gda-connection.h"
diff -ur libgda-3.1.5.orig/libgda/gda-value.c libgda-3.1.5/libgda/gda-value.c
--- libgda-3.1.5.orig/libgda/gda-value.c	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-value.c	2012-06-29 12:53:05.267332537 +0200
@@ -28,10 +28,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <time.h>
-#include <glib/gdate.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gstring.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-blob-op.h>
diff -ur libgda-3.1.5.orig/libgda/gda-value.h libgda-3.1.5/libgda/gda-value.h
--- libgda-3.1.5.orig/libgda/gda-value.h	2008-07-20 11:51:23.000000000 +0200
+++ libgda-3.1.5/libgda/gda-value.h	2012-06-29 12:53:05.267332537 +0200
@@ -27,8 +27,7 @@
 #define __GDA_VALUE_H__
 
 #include <time.h>
-#include <glib/glist.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libxml/tree.h>
 #include <libgda/gda-decl.h>
diff -ur libgda-3.1.5.orig/libgda/sqlite/gda-sqlite.h libgda-3.1.5/libgda/sqlite/gda-sqlite.h
--- libgda-3.1.5.orig/libgda/sqlite/gda-sqlite.h	2008-07-20 11:51:22.000000000 +0200
+++ libgda-3.1.5/libgda/sqlite/gda-sqlite.h	2012-06-29 12:53:05.253999204 +0200
@@ -25,8 +25,7 @@
 #ifndef __GDA_SQLITE_H__
 #define __GDA_SQLITE_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-connection.h>
 
diff -ur libgda-3.1.5.orig/libgda/sqlite/gda-sqlite-provider.c libgda-3.1.5/libgda/sqlite/gda-sqlite-provider.c
--- libgda-3.1.5.orig/libgda/sqlite/gda-sqlite-provider.c	2008-07-20 11:51:22.000000000 +0200
+++ libgda-3.1.5/libgda/sqlite/gda-sqlite-provider.c	2012-06-29 12:53:05.257332538 +0200
@@ -25,7 +25,7 @@
 #include <errno.h>
 #include <string.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <libgda/gda-data-model-array.h>
 #include <libgda/gda-data-model-private.h>
 #include <libgda/gda-parameter-list.h>
diff -ur libgda-3.1.5.orig/libgda-report/DocBook/gda-report-docbook-document.c libgda-3.1.5/libgda-report/DocBook/gda-report-docbook-document.c
--- libgda-3.1.5.orig/libgda-report/DocBook/gda-report-docbook-document.c	2008-07-20 11:51:14.000000000 +0200
+++ libgda-3.1.5/libgda-report/DocBook/gda-report-docbook-document.c	2012-06-29 12:53:05.230665872 +0200
@@ -21,7 +21,7 @@
  */
 
 #include <glib/gi18n-lib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
diff -ur libgda-3.1.5.orig/libgda-report/gda-report-document.c libgda-3.1.5/libgda-report/gda-report-document.c
--- libgda-3.1.5.orig/libgda-report/gda-report-document.c	2008-07-20 11:51:14.000000000 +0200
+++ libgda-3.1.5/libgda-report/gda-report-document.c	2012-06-29 12:53:05.230665872 +0200
@@ -21,7 +21,7 @@
  */
 
 #include <glib/gi18n-lib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
diff -ur libgda-3.1.5.orig/libgda-report/RML/gda-report-rml-document.c libgda-3.1.5/libgda-report/RML/gda-report-rml-document.c
--- libgda-3.1.5.orig/libgda-report/RML/gda-report-rml-document.c	2008-07-20 11:51:14.000000000 +0200
+++ libgda-3.1.5/libgda-report/RML/gda-report-rml-document.c	2012-06-29 12:53:05.230665872 +0200
@@ -21,7 +21,7 @@
  */
 
 #include <glib/gi18n-lib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
diff -ur libgda-3.1.5.orig/libsql/sql_parser.c libgda-3.1.5/libsql/sql_parser.c
--- libgda-3.1.5.orig/libsql/sql_parser.c	2008-07-20 11:51:38.000000000 +0200
+++ libgda-3.1.5/libsql/sql_parser.c	2012-06-29 12:53:05.273999204 +0200
@@ -17,7 +17,7 @@
 #include <stdio.h>
 #include <glib.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <strings.h>
 #include <string.h>
 
diff -ur libgda-3.1.5.orig/providers/bdb/gda-bdb.h libgda-3.1.5/providers/bdb/gda-bdb.h
--- libgda-3.1.5.orig/providers/bdb/gda-bdb.h	2008-07-20 11:51:15.000000000 +0200
+++ libgda-3.1.5/providers/bdb/gda-bdb.h	2012-06-29 12:53:05.273999204 +0200
@@ -25,7 +25,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <db.h>
diff -ur libgda-3.1.5.orig/providers/firebird/gda-firebird-provider.c libgda-3.1.5/providers/firebird/gda-firebird-provider.c
--- libgda-3.1.5.orig/providers/firebird/gda-firebird-provider.c	2008-07-20 11:51:15.000000000 +0200
+++ libgda-3.1.5/providers/firebird/gda-firebird-provider.c	2012-06-29 12:53:05.277332537 +0200
@@ -21,7 +21,7 @@
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-command.h>
 #include <libgda/gda-parameter-list.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <string.h>
 #include "gda-firebird-provider.h"
 #include "gda-firebird-recordset.h"
diff -ur libgda-3.1.5.orig/providers/firebird/gda-firebird-recordset.c libgda-3.1.5/providers/firebird/gda-firebird-recordset.c
--- libgda-3.1.5.orig/providers/firebird/gda-firebird-recordset.c	2008-07-20 11:51:15.000000000 +0200
+++ libgda-3.1.5/providers/firebird/gda-firebird-recordset.c	2012-06-29 12:53:05.273999204 +0200
@@ -27,7 +27,7 @@
 #include <libgda/gda-quark-list.h>
 #include <libgda/gda-parameter-list.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <string.h>
 #include <math.h>
                                                                                                                              
diff -ur libgda-3.1.5.orig/providers/freetds/gda-freetds.h libgda-3.1.5/providers/freetds/gda-freetds.h
--- libgda-3.1.5.orig/providers/freetds/gda-freetds.h	2007-10-01 17:51:37.000000000 +0200
+++ libgda-3.1.5/providers/freetds/gda-freetds.h	2012-06-29 12:53:05.277332537 +0200
@@ -24,7 +24,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <tds.h>
diff -ur libgda-3.1.5.orig/providers/freetds/gda-freetds-message.h libgda-3.1.5/providers/freetds/gda-freetds-message.h
--- libgda-3.1.5.orig/providers/freetds/gda-freetds-message.h	2007-10-01 17:51:37.000000000 +0200
+++ libgda-3.1.5/providers/freetds/gda-freetds-message.h	2012-06-29 12:53:05.277332537 +0200
@@ -24,7 +24,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <tds.h>
diff -ur libgda-3.1.5.orig/providers/freetds/gda-freetds-types.h libgda-3.1.5/providers/freetds/gda-freetds-types.h
--- libgda-3.1.5.orig/providers/freetds/gda-freetds-types.h	2007-10-01 17:51:37.000000000 +0200
+++ libgda-3.1.5/providers/freetds/gda-freetds-types.h	2012-06-29 12:53:05.277332537 +0200
@@ -24,7 +24,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <tds.h>
diff -ur libgda-3.1.5.orig/providers/ibmdb2/gda-ibmdb2.h libgda-3.1.5/providers/ibmdb2/gda-ibmdb2.h
--- libgda-3.1.5.orig/providers/ibmdb2/gda-ibmdb2.h	2007-10-01 17:51:38.000000000 +0200
+++ libgda-3.1.5/providers/ibmdb2/gda-ibmdb2.h	2012-06-29 12:53:05.277332537 +0200
@@ -23,12 +23,11 @@
 #if !defined(__gda_ibmdb2_h__)
 #  define __gda_ibmdb2_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <libgda/gda-value.h>
-#include <glib/gi18n-lib.h>
 #include <sqlcli1.h>
 #include "gda-ibmdb2-provider.h"
 
diff -ur libgda-3.1.5.orig/providers/ibmdb2/gda-ibmdb2-types.h libgda-3.1.5/providers/ibmdb2/gda-ibmdb2-types.h
--- libgda-3.1.5.orig/providers/ibmdb2/gda-ibmdb2-types.h	2007-10-01 17:51:38.000000000 +0200
+++ libgda-3.1.5/providers/ibmdb2/gda-ibmdb2-types.h	2012-06-29 12:53:05.277332537 +0200
@@ -23,7 +23,7 @@
 #ifndef __GDA_IBMDB2_TYPES_H__
 #define __GDA_IBMDB2_TYPES_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <sqlcli1.h>
diff -ur libgda-3.1.5.orig/providers/ldap/gda-ldap.h libgda-3.1.5/providers/ldap/gda-ldap.h
--- libgda-3.1.5.orig/providers/ldap/gda-ldap.h	2007-10-01 17:51:37.000000000 +0200
+++ libgda-3.1.5/providers/ldap/gda-ldap.h	2012-06-29 12:53:05.273999204 +0200
@@ -26,7 +26,7 @@
 #if !defined(__gda_ldap_h__)
 #  define __gda_ldap_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/providers/mdb/gda-mdb.h libgda-3.1.5/providers/mdb/gda-mdb.h
--- libgda-3.1.5.orig/providers/mdb/gda-mdb.h	2008-07-20 11:51:16.000000000 +0200
+++ libgda-3.1.5/providers/mdb/gda-mdb.h	2012-06-29 12:53:05.273999204 +0200
@@ -24,7 +24,7 @@
 #ifndef __GDA_MDB_H__
 #define __GDA_MDB_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/providers/mdb/gda-mdb-provider.c libgda-3.1.5/providers/mdb/gda-mdb-provider.c
--- libgda-3.1.5.orig/providers/mdb/gda-mdb-provider.c	2008-07-20 11:51:16.000000000 +0200
+++ libgda-3.1.5/providers/mdb/gda-mdb-provider.c	2012-06-29 12:53:05.273999204 +0200
@@ -22,7 +22,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gbacktrace.h>
+#include <glib.h>
 #include <virtual/gda-vconnection-data-model.h>
 #include <libgda/gda-data-model-array.h>
 #include <libgda/gda-data-model-private.h>
diff -ur libgda-3.1.5.orig/providers/msql/gda-msql.h libgda-3.1.5/providers/msql/gda-msql.h
--- libgda-3.1.5.orig/providers/msql/gda-msql.h	2007-10-01 17:51:41.000000000 +0200
+++ libgda-3.1.5/providers/msql/gda-msql.h	2012-06-29 12:53:05.273999204 +0200
@@ -22,7 +22,7 @@
 #ifndef __gda_msql_h__
 #define __gda_msql_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/providers/mysql/gda-mysql.h libgda-3.1.5/providers/mysql/gda-mysql.h
--- libgda-3.1.5.orig/providers/mysql/gda-mysql.h	2008-07-20 11:51:16.000000000 +0200
+++ libgda-3.1.5/providers/mysql/gda-mysql.h	2012-06-29 12:53:05.273999204 +0200
@@ -25,7 +25,7 @@
 #ifndef __GDA_MYSQL_H__
 #define __GDA_MYSQL_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/providers/odbc/gda-odbc.h libgda-3.1.5/providers/odbc/gda-odbc.h
--- libgda-3.1.5.orig/providers/odbc/gda-odbc.h	2007-10-01 17:51:41.000000000 +0200
+++ libgda-3.1.5/providers/odbc/gda-odbc.h	2012-06-29 12:53:05.273999204 +0200
@@ -25,7 +25,7 @@
 #if !defined(__gda_odbc_h__)
 #  define __gda_odbc_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
diff -ur libgda-3.1.5.orig/providers/oracle/gda-oracle.h libgda-3.1.5/providers/oracle/gda-oracle.h
--- libgda-3.1.5.orig/providers/oracle/gda-oracle.h	2007-10-01 17:51:38.000000000 +0200
+++ libgda-3.1.5/providers/oracle/gda-oracle.h	2012-06-29 12:53:05.277332537 +0200
@@ -31,7 +31,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
 #include <glib/gi18n-lib.h>
diff -ur libgda-3.1.5.orig/providers/postgres/gda-postgres.h libgda-3.1.5/providers/postgres/gda-postgres.h
--- libgda-3.1.5.orig/providers/postgres/gda-postgres.h	2008-07-20 11:51:16.000000000 +0200
+++ libgda-3.1.5/providers/postgres/gda-postgres.h	2012-06-29 12:53:05.273999204 +0200
@@ -24,7 +24,7 @@
 #ifndef __GDA_POSTGRES_H__
 #define __GDA_POSTGRES_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include "gda-postgres-provider.h"
diff -ur libgda-3.1.5.orig/providers/sybase/gda-sybase.h libgda-3.1.5/providers/sybase/gda-sybase.h
--- libgda-3.1.5.orig/providers/sybase/gda-sybase.h	2007-10-01 17:51:37.000000000 +0200
+++ libgda-3.1.5/providers/sybase/gda-sybase.h	2012-06-29 12:53:05.277332537 +0200
@@ -32,7 +32,7 @@
 #  include <config.h>
 #endif
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #include <ctpublic.h>
diff -ur libgda-3.1.5.orig/testing/gda-test-model-query.c libgda-3.1.5/testing/gda-test-model-query.c
--- libgda-3.1.5.orig/testing/gda-test-model-query.c	2008-07-20 11:51:16.000000000 +0200
+++ libgda-3.1.5/testing/gda-test-model-query.c	2012-06-29 12:53:05.230665872 +0200
@@ -3,7 +3,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <glib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <glib/gi18n-lib.h>
 
diff -ur libgda-3.1.5.orig/testing/gda-test-sql.c libgda-3.1.5/testing/gda-test-sql.c
--- libgda-3.1.5.orig/testing/gda-test-sql.c	2008-07-20 11:51:16.000000000 +0200
+++ libgda-3.1.5/testing/gda-test-sql.c	2012-06-29 12:53:05.230665872 +0200
@@ -3,7 +3,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <glib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <glib/gi18n-lib.h>
 
diff -ur libgda-3.1.5.orig/tools/gda-diagnose.c libgda-3.1.5/tools/gda-diagnose.c
--- libgda-3.1.5.orig/tools/gda-diagnose.c	2008-07-20 11:51:14.000000000 +0200
+++ libgda-3.1.5/tools/gda-diagnose.c	2012-06-29 12:53:05.230665872 +0200
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <glib.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider-extra.h>
diff -ur libgda-3.1.5.orig/tools/gda-inspect-dict-file.c libgda-3.1.5/tools/gda-inspect-dict-file.c
--- libgda-3.1.5.orig/tools/gda-inspect-dict-file.c	2008-07-20 11:51:14.000000000 +0200
+++ libgda-3.1.5/tools/gda-inspect-dict-file.c	2012-06-29 12:53:05.227332539 +0200
@@ -1,6 +1,6 @@
 #include <libgda/libgda.h>
 #include <unistd.h>
-#include <glib/gstdio.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 
 /* options */
diff -ur libgda-3.1.5.orig/tools/gda-sql.c libgda-3.1.5/tools/gda-sql.c
--- libgda-3.1.5.orig/tools/gda-sql.c	2008-07-20 11:55:34.000000000 +0200
+++ libgda-3.1.5/tools/gda-sql.c	2012-06-29 12:53:05.227332539 +0200
@@ -21,12 +21,11 @@
 #include <libgda/libgda.h>
 #include <virtual/libgda-virtual.h>
 #include <glib/gi18n-lib.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <string.h>
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <glib/gstdio.h>
 #include "tools-input.h"
 #include "command-exec.h"
 #include <unistd.h>
