# fix for "Only <glib.h> can be included directly."
--- libgda-3.1.5/libgda/gda-data-model-array.c.orig	2012-06-13 15:37:42.004194678 +0200
+++ libgda-3.1.5/libgda/gda-data-model-array.c	2012-06-13 15:37:56.037473282 +0200
@@ -22,7 +22,7 @@
  */
 
 #include <string.h>
-#include <glib/garray.h>
+#include <glib.h>
 #include <libgda/gda-data-model-array.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-row.h>
--- libgda-3.1.5/libgda/gda-value.c.orig	2012-06-13 15:48:36.818307561 +0200
+++ libgda-3.1.5/libgda/gda-value.c	2012-06-13 15:48:48.691594588 +0200
@@ -28,11 +28,7 @@
 #include <string.h>
 #include <stdlib.h>
 #include <time.h>
-#include <glib/gdate.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gstring.h>
-#include <glib/gi18n-lib.h>
+#include <glib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-blob-op.h>
 #include <libgda/gda-util.h>
--- libgda-3.1.5/libgda/gda-server-provider.c.orig	2012-06-13 15:47:35.408547059 +0200
+++ libgda-3.1.5/libgda/gda-server-provider.c	2012-06-13 15:47:46.915168848 +0200
@@ -22,7 +22,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-server-provider-private.h>
--- libgda-3.1.5/libgda/gda-command.c.orig	2012-06-13 15:17:29.585589796 +0200
+++ libgda-3.1.5/libgda/gda-command.c	2012-06-13 15:17:39.785550017 +0200
@@ -21,9 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-command.h>
 
 GType
--- libgda-3.1.5/libgda/gda-row.h.orig	2012-06-13 15:12:45.593364036 +0200
+++ libgda-3.1.5/libgda/gda-row.h	2012-06-13 15:13:52.876434963 +0200
@@ -27,7 +27,7 @@
 
 #include <glib-object.h>
 #include <libgda/gda-column.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/libgda/gda-data-model-hash.c.orig	2012-06-13 15:38:45.363947574 +0200
+++ libgda-3.1.5/libgda/gda-data-model-hash.c	2012-06-13 15:38:54.823910679 +0200
@@ -23,7 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/ghash.h>
+#include <glib.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-data-model-hash.h>
 #include <libgda/gda-data-model-extra.h>
--- libgda-3.1.5/libgda/gda-data-model-filter-sql.c.orig	2012-06-13 15:38:20.437378120 +0200
+++ libgda-3.1.5/libgda/gda-data-model-filter-sql.c	2012-06-13 15:38:31.830667018 +0200
@@ -21,7 +21,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gdataset.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/libgda/gda-log.c.orig	2012-06-13 15:40:36.426847761 +0200
+++ libgda-3.1.5/libgda/gda-log.c	2012-06-13 15:40:51.310123049 +0200
@@ -24,11 +24,6 @@
 #include <syslog.h>
 #endif
 #include <time.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gutils.h>
-#include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
 
 static GStaticRecMutex gda_mutex = G_STATIC_REC_MUTEX_INIT;
--- libgda-3.1.5/libgda/gda-quark-list.c.orig	2012-06-13 15:42:01.043184423 +0200
+++ libgda-3.1.5/libgda/gda-quark-list.c	2012-06-13 15:42:11.799809139 +0200
@@ -22,10 +22,7 @@
  */
 
 #include <libgda/gda-quark-list.h>
-#include <glib/ghash.h>
-#include <glib/gmem.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 struct _GdaQuarkList {
 	GHashTable *hash_table;
--- libgda-3.1.5/libgda/sqlite/gda-sqlite.h.orig	2012-06-13 12:01:47.728049885 +0200
+++ libgda-3.1.5/libgda/sqlite/gda-sqlite.h	2012-06-13 12:02:09.817963735 +0200
@@ -25,8 +25,7 @@
 #ifndef __GDA_SQLITE_H__
 #define __GDA_SQLITE_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <libgda/gda-value.h>
 #include <libgda/gda-connection.h>
 
--- libgda-3.1.5/libgda/gda-config.h.orig	2012-06-13 15:13:15.906579148 +0200
+++ libgda-3.1.5/libgda/gda-config.h	2012-06-13 15:14:11.616361879 +0200
@@ -23,7 +23,7 @@
 #ifndef __GDA_CONFIG_H__
 #define __GDA_CONFIG_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-data-model.h>
 #include <libgda/gda-parameter.h>
 
--- libgda-3.1.5/libgda/gda-server-operation.c.orig	2012-06-13 15:47:02.018677280 +0200
+++ libgda-3.1.5/libgda/gda-server-operation.c	2012-06-13 15:47:14.061963644 +0200
@@ -21,7 +21,7 @@
  */
 
 #include <stdlib.h>
-#include <glib/gmessages.h>
+#include <glib.h>
 #include <libgda/gda-marshal.h>
 #include <libgda/gda-dict.h>
 #include <libgda/gda-dict-type.h>
--- libgda-3.1.5/libgda/gda-value.h.orig	2012-06-13 15:12:33.963409394 +0200
+++ libgda-3.1.5/libgda/gda-value.h	2012-06-13 15:13:46.156461173 +0200
@@ -27,8 +27,7 @@
 #define __GDA_VALUE_H__
 
 #include <time.h>
-#include <glib/glist.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <libxml/tree.h>
 #include <libgda/gda-decl.h>
--- libgda-3.1.5/libgda/gda-util.c.orig	2012-06-13 15:48:09.021749299 +0200
+++ libgda-3.1.5/libgda/gda-util.c	2012-06-13 15:48:21.341701253 +0200
@@ -27,10 +27,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <glib/gfileutils.h>
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <libsql/sql_parser.h>
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-log.h>
--- libgda-3.1.5/libgda/gda-init.c.orig	2012-06-13 15:40:13.713603009 +0200
+++ libgda-3.1.5/libgda/gda-init.c	2012-06-13 15:40:22.516902010 +0200
@@ -17,7 +17,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <gmodule.h>
 #include <libgda/libgda.h>
 #include <glib/gi18n-lib.h>
--- libgda-3.1.5/libgda/gda-util.h.orig	2012-06-13 15:15:05.799483896 +0200
+++ libgda-3.1.5/libgda/gda-util.h	2012-06-13 15:15:15.329446731 +0200
@@ -24,8 +24,7 @@
 #ifndef __GDA_UTIL_H__
 #define __GDA_UTIL_H__
 
-#include <glib/ghash.h>
-#include <glib/glist.h>
+#include <glib.h>
 #include "gda-parameter.h"
 #include "gda-row.h"
 #include "gda-connection.h"
--- libgda-3.1.5/libgda/gda-quark-list.h.orig	2012-06-13 15:11:18.110371889 +0200
+++ libgda-3.1.5/libgda/gda-quark-list.h	2012-06-13 15:45:51.705618170 +0200
@@ -24,8 +24,7 @@
 #ifndef __GDA_QUARK_LIST_H__
 #define __GDA_QUARK_LIST_H__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/libgda/gda-log.h.orig	2012-06-13 15:11:04.947089892 +0200
+++ libgda-3.1.5/libgda/gda-log.h	2012-06-13 15:11:48.903585127 +0200
@@ -20,8 +20,7 @@
 #if !defined(__gda_log_h__)
 #  define __gda_log_h__
 
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <time.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/libgda/gda-column.c.orig	2012-06-13 15:17:04.739020031 +0200
+++ libgda-3.1.5/libgda/gda-column.c	2012-06-13 15:17:19.042297582 +0200
@@ -23,8 +23,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include <glib/gmessages.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgda/gda-column.h>
 #include <string.h>
 #include "gda-marshal.h"
--- libgda-3.1.5/libgda/gda-config.c.orig	2012-06-13 15:17:49.435512380 +0200
+++ libgda-3.1.5/libgda/gda-config.c	2012-06-13 15:18:10.932095209 +0200
@@ -38,7 +38,6 @@
 #include <sys/stat.h>
 #ifdef HAVE_FAM
 #include <fam.h>
-#include <glib/giochannel.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
--- libgda-3.1.5/libgda/gda-column.h.orig	2012-06-13 15:12:59.336643771 +0200
+++ libgda-3.1.5/libgda/gda-column.h	2012-06-13 15:13:58.969744535 +0200
@@ -28,7 +28,7 @@
 
 #include <glib-object.h>
 #include <libgda/gda-value.h>
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-decl.h>
 
 G_BEGIN_DECLS
--- libgda-3.1.5/providers/bdb/gda-bdb.h.orig	2012-06-13 15:49:21.074801624 +0200
+++ libgda-3.1.5/providers/bdb/gda-bdb.h	2012-06-13 15:49:30.218099299 +0200
@@ -25,8 +25,7 @@
 #if defined(HAVE_CONFIG_H)
 #endif
 
-#include <glib/gmacros.h>
-#include <glib/gi18n-lib.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include <db.h>
 
--- libgda-3.1.5/providers/mysql/gda-mysql.h.orig	2012-06-13 16:13:22.575846417 +0200
+++ libgda-3.1.5/providers/mysql/gda-mysql.h	2012-06-13 16:13:33.602470079 +0200
@@ -25,10 +25,9 @@
 #ifndef __GDA_MYSQL_H__
 #define __GDA_MYSQL_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
-#include <glib/gi18n-lib.h>
 #include <libgda/gda-value.h>
 #include "gda-mysql-provider.h"
 
--- libgda-3.1.5/providers/sybase/gda-sybase.h.orig	2012-06-13 16:15:26.898694889 +0200
+++ libgda-3.1.5/providers/sybase/gda-sybase.h	2012-06-13 16:15:38.218650740 +0200
@@ -32,8 +32,7 @@
 #  include <config.h>
 #endif
 
-#include <glib/gmacros.h>
-#include <glib/gi18n-lib.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include <ctpublic.h>
 #include <cspublic.h>
--- libgda-3.1.5/providers/postgres/gda-postgres.h.orig	2012-06-13 16:14:44.082195207 +0200
+++ libgda-3.1.5/providers/postgres/gda-postgres.h	2012-06-13 16:14:53.112159990 +0200
@@ -24,8 +24,7 @@
 #ifndef __GDA_POSTGRES_H__
 #define __GDA_POSTGRES_H__
 
-#include <glib/gmacros.h>
-#include <glib/gi18n-lib.h>
+#include <glib.h>
 #include <libgda/gda-server-provider.h>
 #include "gda-postgres-provider.h"
 #include "gda-postgres-recordset.h"
--- libgda-3.1.5/providers/odbc/gda-odbc.h.orig	2012-06-13 16:14:18.495628328 +0200
+++ libgda-3.1.5/providers/odbc/gda-odbc.h	2012-06-13 16:14:29.272252967 +0200
@@ -25,9 +25,8 @@
 #if !defined(__gda_odbc_h__)
 #  define __gda_odbc_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <glib-object.h>
-#include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider.h>
 #if defined(__CYGWIN__)
 #  define WIN32_LEAN_AND_MEAN
--- libgda-3.1.5/providers/mdb/gda-mdb.h.orig	2012-06-13 16:12:58.035942122 +0200
+++ libgda-3.1.5/providers/mdb/gda-mdb.h	2012-06-13 16:13:07.635904681 +0200
@@ -24,10 +24,9 @@
 #ifndef __GDA_MDB_H__
 #define __GDA_MDB_H__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
-#include <glib/gi18n-lib.h>
 #include <libgda/gda-value.h>
 #include <mdbtools.h>
 #include "gda-mdb-provider.h"
--- libgda-3.1.5/providers/mdb/gda-mdb-provider.c.orig	2012-06-13 16:12:35.276030887 +0200
+++ libgda-3.1.5/providers/mdb/gda-mdb-provider.c	2012-06-13 16:12:46.222654860 +0200
@@ -22,11 +22,10 @@
  */
 
 #include <stdlib.h>
-#include <glib/gbacktrace.h>
+#include <glib.h>
 #include <virtual/gda-vconnection-data-model.h>
 #include <libgda/gda-data-model-array.h>
 #include <libgda/gda-data-model-private.h>
-#include <glib/gi18n-lib.h>
 #include <libgda/gda-server-provider-extra.h>
 #include <libgda/gda-parameter-list.h>
 #include "gda-mdb.h"
--- libgda-3.1.5/providers/ldap/gda-ldap.h.orig	2012-06-13 16:18:34.837961923 +0200
+++ libgda-3.1.5/providers/ldap/gda-ldap.h	2012-06-13 16:18:45.984585117 +0200
@@ -26,10 +26,9 @@
 #if !defined(__gda_ldap_h__)
 #  define __gda_ldap_h__
 
-#include <glib/gmacros.h>
+#include <glib.h>
 #include <libgda/gda-connection.h>
 #include <libgda/gda-server-provider.h>
-#include <glib/gi18n-lib.h>
 #include <libgda/gda-value.h>
 #include "gda-ldap-provider.h"
 /* TODO: we use the old API.  We should update this to the new API. */
--- libgda-3.1.5/libgda/gda-command.h.orig	2012-06-13 16:27:03.535977993 +0200
+++ libgda-3.1.5/libgda/gda-command.h	2012-06-13 16:27:15.849263303 +0200
@@ -25,8 +25,7 @@
 #define __GDA_COMMAND_H__
 
 #include <glib-object.h>
-#include <glib/gmacros.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
