--- libgda-3.0.1/providers/firebird/gda-firebird-provider.c.orig	2007-04-24 23:13:13.000000000 +0200
+++ libgda-3.0.1/providers/firebird/gda-firebird-provider.c	2007-05-31 21:20:50.706932530 +0200
@@ -24,6 +24,7 @@
 #include <glib/gi18n-lib.h>
 #include <libgda/gda-data-model-array.h>
 #include <libgda/gda-data-model-private.h>
+#include <libgda/gda-parameter-list.h>
 #include <libgda/gda-command.h>
 #include <glib/gprintf.h>
 #include <string.h>
@@ -309,7 +310,7 @@
 		
 		/* Find parameters */
 		if (params)
-			par = gda_parameter_list_find (params, "systables");
+			par = gda_parameter_list_find_param (params, "systables");
 
 		/* Initialize parameter values */
 		if (par)
@@ -343,7 +344,7 @@
 				GValue *tmpval;
 
 				/* Get name of table */
-				row = (GdaRow *) gda_data_model_get_row (recmodel, i);
+				row = (GdaRow *) gda_data_model_row_get_row (GDA_DATA_MODEL_ROW(recmodel), i, NULL);
 				value = gda_row_get_value (row, 0);
 				value_list = g_list_append (value_list, gda_value_copy (value));
 
@@ -624,7 +625,7 @@
 	g_return_val_if_fail (GDA_IS_CONNECTION (cnc), NULL);
 	g_return_val_if_fail (params != NULL, NULL);
 	
-	par = gda_parameter_list_find (params, "name");
+	par = gda_parameter_list_find_param (params, "name");
 	if (!par) {
 		gda_connection_add_event_string (cnc,
 				_("Table name is needed but none specified in parameter list"));
@@ -659,7 +660,7 @@
 			
 			for (i = 0; i < gda_data_model_get_n_rows (GDA_DATA_MODEL (reclist->data)); i++) {
 				value_list = NULL;
-				row = (GdaRow *) gda_data_model_get_row (GDA_DATA_MODEL (reclist->data), i);
+				row = (GdaRow *) gda_data_model_row_get_row (GDA_DATA_MODEL_ROW (GDA_DATA_MODEL (reclist->data)), i, NULL);
 				
 				/* Set field metdata for row, then append to recordset */
 				value_list = fb_set_field_metadata (row);
