--- libgda-3.1.5/providers/mdb/gda-mdb-provider.c~	2013-11-30 22:13:45.023125651 +0100
+++ libgda-3.1.5/providers/mdb/gda-mdb-provider.c	2013-11-30 22:25:22.513751601 +0100
@@ -370,7 +370,7 @@
         case MDB_NUMERIC : return GDA_TYPE_NUMERIC;
         case MDB_OLE : return GDA_TYPE_BINARY;
         case MDB_REPID : return GDA_TYPE_BINARY;
-        case MDB_SDATETIME : return GDA_TYPE_TIMESTAMP;
+        case MDB_DATETIME : return GDA_TYPE_TIMESTAMP;
         case MDB_TEXT : return G_TYPE_STRING;
         }
 
